hawkular-inventory-parent/hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/oldmodel/Metric.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2014-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.inventory.api.oldmodel;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import org.hawkular.inventory.paths.CanonicalPath;
+import org.hawkular.inventory.paths.SegmentType;
+
+import io.swagger.annotations.ApiModel;
+
+/**
+ * Metric describes a single metric that is sent out from a feed. Each metric has a unique ID and a type. Metrics live
+ * in an environment and can be "incorporated" by {@link Resource resources} (surprisingly, many resources can
+ * incorporate a single metric).
+ *
+ * @author Lukas Krejci
+ * @since 0.0.1
+ */
+@ApiModel(description = "A metric represents a monitored \"quality\". Its metric type specifies the unit in which" +
```

Comment:
Side note.
In Hawkular Alerting we have migrated from Swagger annotations into Hawkular custom annotations which are more flexible for defining our api (and also it works better with vertx handlers).
It is in my todo to put them in commons but I see it can be useful for there too.